server/src/test/java/org/candlepin/model/ProductTest.java

```diff
@@ -88,14 +89,44 @@
             new Branding(null, "eng_prod_id_6", "eng_prod_name_6", "OS")
         ).collect(Collectors.toSet());
 
+        Set<Product> providedP1 = Stream.of(
+            new Product("ak1", "providedProduct1", "varient1", "version1",
+            "arch1" , "type1", null, null),
+            new Product("ak2", "providedProduct2", "varient2", "version2",
+            "arch2", "type2", null, null)
+        ).collect(Collectors.toSet());
+
+        Set<Product> providedP2 = Stream.of(
+            new Product("ak3", "providedProduct3", "varient3", "version3",
+            "arch3" , "type3", null, null),
+            new Product("ak4", "providedProduct4", "varient4", "version4",
+            "arch4" , "type4", null, null)
+        ).collect(Collectors.toSet());
+
+        Set<Product> derivedProvidedP1 = Stream.of(
+            new Product("dp1", "derivedProvidedP1", "varient1", "version1",
+            "arch1", "type1", null, null),
+            new Product("dp2", "derivedProvidedP2", "varient2", "version2",
+            "arch2", "type2", null, null)
+        ).collect(Collectors.toSet());
+
+        Set<Product> derivedProvidedP2 = Stream.of(
+            new Product("dp3", "derivedProvidedP3", "varient3", "version3",
+            "arch3", "type3", null, null),
+            new Product("dp4", "derivedProvidedP4", "varient4", "version4",
+            "arch2", "type2", null, null)
+        ).collect(Collectors.toSet());
+
         return Stream.of(
             new Object[] { "Id", "test_value", "alt_value" },
             new Object[] { "Name", "test_value", "alt_value" },
             new Object[] { "Multiplier", 1234L, 4567L },
             new Object[] { "Attributes", attributes1, attributes2 },
             new Object[] { "ProductContent", productContent1, productContent2 },
             new Object[] { "DependentProductIds", Arrays.asList("1", "2", "3"), Arrays.asList("4", "5") },
-            new Object[] { "Branding", brandings1, brandings2 }
+            new Object[] { "Branding", brandings1, brandings2 },
+            new Object[] { "ProvidedProducts", providedP1, providedP2},
+            new Object[] { "DerivedProvidedProducts", derivedProvidedP1, derivedProvidedP2}
```

Comment:
Inconsistent whitespace at the end of these last two lines. Should be a space between P2 and the curly brace.

Reply:
fixed.